### PR TITLE
Adding cancel polling component

### PR DESCRIPTION
**Pull request by neelSharma12**
### New
<new>

-   For QR code payment methods, the QR code screen now has a **Cancel** button instead of a **x** button in the corner for the shopper to cancel the payment.

</new>

### CODE REVIEW

File: Adyen/Core/ToolBar/CancellableToolBar.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import UIKit
+
+@_spi(AdyenInternal)
+public class CancellableToolBar: ModalToolbar {
```
Comment (descorp) [inline Adyen/Core/ToolBar/CancellableToolBar.swift]: Could be marked `final`

File: Adyen/Core/ToolBar/CancellableToolBar.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        self.style = style
+        self.title = title
+        super.init(title: title, style: style)
+    }
+
+    private func customizeCancelButton() -> UIButton {
+        let button = UIButton(type: UIButton.ButtonType.system)
+        let cancelText = Bundle(for: UIApplication.self).localizedString(forKey: "Cancel", value: nil, table: nil)
+        button.setTitle(cancelText, for: .normal)
+        button.titleLabel?.font = .systemFont(ofSize: 17.0)
```
Comment (descorp) [inline Adyen/Core/ToolBar/CancellableToolBar.swift]: Is this necessary? What font is it by default?

Comment (neelSharma12) [reply Adyen/Core/ToolBar/CancellableToolBar.swift]: Yes this is necessary.

Comment (descorp) [inline Adyen/Core/ToolBar/CancellableToolBar.swift]: This is not very "accessibility" friendly.
Better use one of [preferred fonts](https://www.iosfontsizes.com/) fonts `.preferredFont(forTextStyle: .body)` or create custom one in Adyen/UI/Styles/CoreFonts.swift

File: UITests/Components/QRCode/QRCodeActionComponentUITests.swift [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 @_spi(AdyenInternal) @testable import Adyen
 @testable import AdyenActions
+@testable import AdyenDropIn
```
Comment (descorp) [inline UITests/Components/QRCode/QRCodeActionComponentUITests.swift:9]: Minor:
This is no longer needed :)

Comment (descorp) [inline UITests/Components/QRCode/QRCodeActionComponentUITests.swift:9]: Minor:
This is no longer needed :)

Comment (neelSharma12) [reply UITests/Components/QRCode/QRCodeActionComponentUITests.swift:9]: This is needed for ```WrapperViewController```

Comment (neelSharma12) [reply UITests/Components/QRCode/QRCodeActionComponentUITests.swift:9]: This is needed for ```WrapperViewController```

File: Adyen/Core/ToolBar/CancellableToolBar.swift (hunk 1 above)
Comment (neelSharma12) [reply Adyen/Core/ToolBar/CancellableToolBar.swift]: Made it final 371694aca888a0dbd865e7232c47d8275294ea30

File: Adyen/Core/ToolBar/CancellableToolBar.swift (hunk 2 above)
Comment (neelSharma12) [reply Adyen/Core/ToolBar/CancellableToolBar.swift]: Changed it. 371694aca888a0dbd865e7232c47d8275294ea30

**Review by erenbesel — APPROVED** [review]

**Pull request merged**